src/dataSources/Null.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+var Base = require('../lib/Base');
+
+var Null = Base.extend('Null', {
+    isNullObject: function() {
+        return true;
+    },
+    getProperty: function getProperty(propName) {
+        if (propName in this) {
+            return this[propName];
+        }
+
+        if (this.dataSource) {
+            return getProperty.call(this.dataSource, propName);
+        }
+    },
+    getDataIndex: function(y) {
+        return 0;
+    },
+    getRow: function() {
+        return null;
+    },
+    findRow: function(columnName, value) {
```

Comment:
@Dwaynekj Unfortunately having this null method defined here (just to serve as an "interface definition") is problematic -- if this file is to serve as a base, as we discussed. The problem is the forwarding routines _depend on such methods being undefined_ and if they are defined here it will break the forwarding mechanism.

To be clear, by "forwarding routines" I mean this (from hyper-analytics/js/Base.js):

```javascript
    findRow: function() {
        if (this.dataSource) {
            return this.dataSource.findRow.apply(this.dataSource, arguments);
        }
    },
```

We could resolve this issue, although not performant, like this:

```javascript
    findRow: function() {
        if (this.dataSource && this.dataSource.findRow === this.findRow) {
            return this.dataSource.findRow.apply(this.dataSource, arguments);
        }
    },
```

Simpler and more performant would be to comment out all the null methods, or at least the ones for which an implementation in the data source "class" is optional. These would include all the ones for which there are forwarding functions. Obviously, methods _required_ to be implemented can be left uncommented _though of course they will never be called_ so might as well comment them out as well. We can solve this probelm simply by not depending on code for this purpose (interface definition). Just leave them as comments or better yet IMHO put them in an external document and link to that document.